addons/modules/functions/fnc_moduleCreateMinefield.sqf

```diff
@@ -21,31 +21,65 @@ private _minesCache = +(uiNamespace getVariable QGVAR(minesCache));
 _minesCache params ["_configNames", "_displayNames"];
 
 [LSTRING(ModuleCreateMinefield), [
-    ["VECTOR", LSTRING(ModuleCreateMinefield_MineArea), [100, 100]],
-    ["COMBO", LSTRING(ModuleCreateMinefield_MineType), [_configNames, _displayNames, 0]],
-    ["TOOLBOX", LSTRING(ModuleCreateMinefield_MineDensity), [2, 1, 5, [ELSTRING(common,VeryLow), ELSTRING(common,Low), ELSTRING(common,Medium), ELSTRING(common,High), ELSTRING(common,VeryHigh)]]]
+    [
+        "VECTOR",
+        LSTRING(ModuleCreateMinefield_MineArea),
+        [100, 100]
+    ],
+    [
+        "COMBO",
+        LSTRING(ModuleCreateMinefield_MineType),
+        [_configNames, _displayNames, 0]
+    ],
+    [
+        "TOOLBOX",
+        LSTRING(ModuleCreateMinefield_MineDensity),
+        [2, 1, 5, [
+            ELSTRING(common,VeryLow),
+            ELSTRING(common,Low),
+            ELSTRING(common,Medium),
+            ELSTRING(common,High),
+            ELSTRING(common,VeryHigh)
+        ]]
+    ],
+    [
+        "TOOLBOX:YESNO",
+        [LSTRING(ModuleCreateMinefield_RandomPlacement), LSTRING(ModuleCreateMinefield_RandomPlacement_Tooltip)],
+        false
+    ]
 ], {
-    params ["_dialogValues", "_position"];
-    _dialogValues params ["_area", "_mineType", "_mineDensity"];
-    _area params ["_areaWidth", "_areaHeight"];
+    params ["_values", "_position"];
+    _values params ["_area", "_type", "_density", "_randomize"];
+    _area params ["_width", "_height"];
 
-    private _createdMines = [];
+    private _mines = [];
 
     // Determine spacing based on density
-    private _mineSpacing = -5 * _mineDensity + 30;
-    private _topLeftPos = _position vectorAdd [-_areaWidth / 2, -_areaHeight / 2, 0];
-
-    // Create minefield of given area
-    for "_i" from _mineSpacing / 2 to _areaWidth step _mineSpacing do {
-        for "_j" from _mineSpacing / 2 to _areaHeight step _mineSpacing do {
-            private _mine = createVehicle [_mineType, [0, 0, 0], [], 0, "CAN_COLLIDE"];
-            _mine setPos (_topLeftPos vectorAdd [_i, _j, 0]);
-            _createdMines pushBack _mine;
+    private _spacing = 30 + _density * -5;
```

Comment:
I would suggest the following, because the very low density feels like a medium/high density minefield.
```suggestion
    private _spacing = 60 + _density * -10;
```